### PR TITLE
Room List Page Support

### PR DESCRIPTION
**Pull request by hieuphq**
## Description
This update focuses on managing rooms

## Checklists
- [x] New API for Room List: retrieve and display the list of available rooms.
- [x] Index Page: showcase the room list.
- [x] New Room Page for Games: show the gaming world with a room page, requiring the room_token for access
- [x] WebSocket Logic: the WebSocket logic by connecting games through the game_token

### CODE REVIEW

File: server/src/actors/room_manager_actor.rs [hunk 1]
```diff
@@ -148,3 +148,41 @@ impl Handler<JoinRoom> for RoomManagerActor {
...
+    pub max_players: u32,
+    pub time_limit_seconds: u32,
+    pub token: String,
+}
+
+impl Handler<ListRooms> for RoomManagerActor {
+    type Result = MessageResult<ListRooms>;
+
+    fn handle(&mut self, _msg: ListRooms, _ctx: &mut Self::Context) -> Self::Result {
+        let rooms: Vec<&Room> = self.rooms.values().collect();
```
Comment (unrealhoang) [inline server/src/actors/room_manager_actor.rs]: don't need to collect here

File: server/src/controllers/api.rs [hunk 2]
```diff
@@ -16,27 +17,52 @@ pub fn socket_handler(
...
+            ),
+            Err(err) => Err(actix_web::error::ErrorBadRequest(err.to_string())),
+        }
     } else {
         Err(actix_web::error::ErrorBadRequest("Invalid API Key"))
     }
 }
 
+#[derive(Debug, Deserialize)]
+pub struct InspectatorString {
```
Comment (unrealhoang) [inline server/src/controllers/api.rs]: `Spectator`?

**Review by unrealhoang — APPROVED** [review]
lgtm with nits

File: server/src/actors/room_manager_actor.rs (hunk 1 above)
Comment (hieuphq) [reply server/src/actors/room_manager_actor.rs]: agreed

File: server/src/controllers/api.rs (hunk 2 above)
Comment (hieuphq) [reply server/src/controllers/api.rs]: updated

**Review by unrealhoang — APPROVED** [review]
lgtm

**Pull request merged**